src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -82,27 +82,24 @@ def _wait(self, completions):
 
         Waits for writes to be *persistent* but not *effective*.
         """
-        done = False
-
-        while done is False:
-            done = self._oremote("wait", completions) == []
```

Comment:
Ok, after looking at the code, `self._oremote("wait", completions) == []` is always False, because `wait()` returns a boolean, not a list.

Reply:
Not in the case of the Ansible Orchestrator, that returned the list of completions objects with operations not finished or an empty list if all operations finish.
As you say, and it is stated in the documentation, the orchestrator wait method MUST return a boolean. So i have fixed this in the ansible orchestrator.
However, this does not explain the weird behavior when a returned empty list is deemed "not False".